### DISCUSSION TITLE
Eliminate render-blocking CSS

### DISCUSSION THREAD

**Discussion by ciferkey** [mmistakes/minimal-mistakes#2838] [Ideas]
## Environment

- Minimal Mistakes version: 4.22.0
- Ruby gem or remote theme version: 4.22.0
- Jekyll version: [3.9.0](https://pages.github.com/versions/)
- Git repository URL: https://github.com/ciferkey/ciferkey.github.com
- GitHub Pages hosted (if yes provide URL to site): https://blog.matthewbrunelle.com
- Operating system: NixOS

## Expected behavior

The theme does not use render blocking style sheets.

## Steps to reproduce the behavior

Create a blog with the theme. Run a Google Lighthouse report of the site. 

It will [flag it with render blocking style sheets](https://web.dev/render-blocking-resources/?utm_source=lighthouse&utm_medium=lr) for the following files:

 * …css/main.css
 * …css/all.min.css(cdn.jsdelivr.net)

## Other

I believe these are the lines where the style sheets are loaded in:
https://github.com/mmistakes/minimal-mistakes/blob/7bb68edf0353a72e82a4c39c6b153eca6721a1eb/_includes/head.html#L17-L18

It looks like the project is using SASS but it also looks like its converted to CSS and then the CSS is included. 

One option might be to use [Scssify to inline critical CSS](https://joshvogt.co/blog/critical-path-jekyll/) but I'm not sure if this is compatible with how the project is setup. 

## Previous Issues
Here is all the previous discussion I could find:
* https://github.com/mmistakes/minimal-mistakes/issues/2704 was opened and closed 2 days later by the OP with no discussion

**Comment by iBug** [comment]
It's going to add a lot of complexity and not easily feasible with the current setup. You can give it a try if you have the right ideas but I won't consider this "an issue" of the theme.

Also, it's a bad idea to try to apply whatever "optimization guide" blindly without understanding them (and your site).
